### PR TITLE
Better BPINN ode Solver

### PR DESCRIPTION
**Pull request by AstitvaAggarwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by AstitvaAggarwal** [comment]
@ChrisRackauckas @Vaibhavdixit02 for the new formulations, merging of this PR first and PR #842 later is recommended.

**Review by Vaibhavdixit02 — APPROVED** [review]
LGTM

File: test/bpinnexperimental.jl [hunk 1]
```diff
@@ -0,0 +1,140 @@
+using Test, MCMCChains
```
Comment (ChrisRackauckas) [inline test/bpinnexperimental.jl:1]: this new test file is not actually ran.

File: test/BPINN_Tests.jl [hunk 2]
```diff
@@ -209,12 +212,12 @@ end
...
 
     # estimated parameters(lux chain)
-    param1 = mean(i[62] for i in fhsampleslux22[1000:1500])
+    param1 = mean(i[62] for i in fhsampleslux22[1000:length(fhsampleslux22)])
     @test abs(param1 - p) < abs(0.3 * p)
 
     #-------------------------- solve() call 
     # (lux chain)
-    @test mean(abs.(physsol2 .- sol3lux_pestim.ensemblesol[1])) < 0.15
+    @test mean(abs.(physsol2 .- pmean(sol3lux_pestim.ensemblesol[1]))) < 0.15
```
Comment (ChrisRackauckas) [inline test/BPINN_Tests.jl:220]: `pmean` typo?

File: test/BPINN_PDEinvsol_tests.jl [hunk 3]
```diff
@@ -59,17 +59,17 @@ Random.seed!(100)
...
+    #     dataset = [dataset, nothing])
+
+    # ahmc_bayesian_pinn_pde(pde_system,
+    #     discretization;
+    #     draw_samples = 1500,
+    #     bcstd = [0.05],
+    #     phystd = [0.01], l2std = [0.01],
+    #     priorsNNw = (0.0, 1.0),
+    #     saveats = [1 / 50.0],
+    #     param = [LogNormal(6.0, 0.5)])
```
Comment (ChrisRackauckas) [inline test/BPINN_PDEinvsol_tests.jl:72]:
```suggestion
```

File: src/collocated_estim.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# suggested extra loss function for ODE solver case
+function L2loss2(Tar::LogTargetDensity, θ)
```
Comment (ChrisRackauckas) [inline src/collocated_estim.jl:2]: why is this a separate file/

**Comment by ChrisRackauckas** [comment]
My understanding is that the proposed changes from the last review were not implemented and this was just rebased? Is there a plan to finish this?

File: test/BPINN_Tests.jl (hunk 2 above)
Comment (AstitvaAggarwal) [reply test/BPINN_Tests.jl:220]: nope, the mean is required as the solution's standard deviation are different at domain points, sometimes keeping this uncertainty is too large causing the tests to fail. so i just took the means for testing

File: src/collocated_estim.jl (hunk 4 above)
Comment (AstitvaAggarwal) [reply src/collocated_estim.jl]: fixed

File: test/bpinnexperimental.jl (hunk 1 above)
Comment (AstitvaAggarwal) [reply test/bpinnexperimental.jl]: yeah, fixed

**Comment by AstitvaAggarwal** [comment]
This is ready @ChrisRackauckas the other test failures seem unrelated. Let me know if changes are needed.

**Pull request merged**